src/Agent.Sdk/Util/ParentProcessUtil/WindowsParentProcessUtil.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Agent.Sdk.Util.ParentProcessUtil
+{
+    public enum ParentProcessNames { Pwsh, Powershell };
+
+    public static class WindowsParentProcessUtil
+    {
+        public static (bool, Dictionary<string, string>) IsParentProcess(bool useInteropToFindParentProcess, params ParentProcessNames[] processNames)
+        {
+            var processList = new List<Process>();
+
+            try
+            {
+                (processList, var telemetry) = GetProcessList(Process.GetCurrentProcess(), useInteropToFindParentProcess);
+
+                bool isProcessRunningInPowerShell = processList.Exists(process => IsProcessPowershell(process, processNames));
+
+                return (isProcessRunningInPowerShell, telemetry);
+            }
+            finally
+            {
+                foreach (var process in processList)
+                {
+                    process.Dispose();
+                }
+            }
+        }
+
+        internal static (List<Process>, Dictionary<string, string>) GetProcessList(Process process, bool useInterop)
+        {
+            var telemetry = new Dictionary<string, string>();
+            var processList = new List<Process>() { process };
+            const int maxSearchDepthForProcess = 10;
+
+            while (processList.Count < maxSearchDepthForProcess)
+            {
+                Process lastProcess = processList.Last();
```

Comment:
this will throw if processlist is empty.  use LastOrDefault instead?

Reply:
The list if prepopulated with the current process, so it's not expected to be empty